barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -2638,76 +2550,73 @@ void AvmTraceBuilder::op_call([[maybe_unused]] uint8_t indirect,
                               [[maybe_unused]] uint32_t success_offset,
                               [[maybe_unused]] uint32_t function_selector_offset)
 {
-    // pc++;
     auto clk = static_cast<uint32_t>(main_trace.size()) + 1;
     const ExternalCallHint& hint = execution_hints.externalcall_hints.at(external_call_counter);
-    // We can load up to 4 things per row
-    auto register_order = std::array{ IntermRegister::IA, IntermRegister::IB, IntermRegister::IC, IntermRegister::ID };
-    // Constrain gas cost
+
     gas_trace_builder.constrain_gas_for_external_call(
         clk, static_cast<uint32_t>(hint.l2_gas_used), static_cast<uint32_t>(hint.da_gas_used));
-    // Indirect is ZEROTH, SECOND and FOURTH bit  COME BACK TO MAKING THIS ALL SUPPORTED
-    auto read_ind_gas_offset =
-        mem_trace_builder.indirect_read_and_load_from_memory(call_ptr, clk, IndirectRegister::IND_A, gas_offset);
-    auto read_ind_args_offset =
-        mem_trace_builder.indirect_read_and_load_from_memory(call_ptr, clk, IndirectRegister::IND_C, args_offset);
-
-    std::vector<uint32_t> first_row_load = {
-        uint32_t(read_ind_gas_offset.val),
-        addr_offset,
-        uint32_t(read_ind_args_offset.val),
-    };
-    std::vector<FF> first_row_values = {};
-    for (uint32_t j = 0; j < first_row_load.size(); j++) {
-        // We just read and load to set up the constraints, we dont actually use these values for now.
-        // info("Register order ", register_order[j]);
-        auto mem_read = mem_trace_builder.read_and_load_from_memory(
-            call_ptr, clk, register_order[j], first_row_load[j], AvmMemoryTag::FF, AvmMemoryTag::U0);
-        first_row_values.emplace_back(mem_read.val);
-    }
+    // We can load up to 4 things per row
+    auto [resolved_gas_offset,
+          resolved_addr_offset,
+          resolved_args_offset,
+          resolved_args_size,
+          resolved_ret_offset,
+          resolved_success_offset] =
+        resolve_indirects<6>(indirect, { gas_offset, addr_offset, args_offset, args_size, ret_offset, success_offset });
+
+    // Should read the address next to read_gas as well (tuple of gas values)
+    auto read_gas = constrained_read_from_memory<IntermRegister::IA>(
+        call_ptr, clk, resolved_gas_offset, AvmMemoryTag::FF, AvmMemoryTag::U0);
+    auto read_addr = constrained_read_from_memory<IntermRegister::IC>(
+        call_ptr, clk, resolved_addr_offset, AvmMemoryTag::FF, AvmMemoryTag::U0);
+    auto read_args = constrained_read_from_memory<IntermRegister::ID>(
+        call_ptr, clk, resolved_args_offset, AvmMemoryTag::FF, AvmMemoryTag::U0);
+    // auto read_args_size = constrained_read_from_memory<IntermRegister::ID>(
```

Comment:
Why is this commented out?

Reply:
The read_args_size isnt used for now.

I'll need to go through and fix this op_call in another PR, as the test is also commented in an earlier pr